modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts

```diff
@@ -150,6 +155,80 @@ export default class Ecdsa {
     return participants;
   }
 
+  /**
+   * Derive shares for a BIP-32 subkey.
+   * @param {PShare} The user's p-share.
+   * @param {NShare[]} The n-shares received from the other participants.
+   * @param {string} The BIP-32 path to derive.
+   * @returns {SubkeyShare} Returns the private x-share and n-shares to
+   * be distributed to participants at their corresponding index.
+   */
+  keyDerive(pShare: PShare, nShares: NShare[], path: string): SubkeyShare {
+    const yValues = [pShare, ...nShares].map((share) => hexToBigInt(share.y));
+    const y = yValues.reduce((partial, share) => Ecdsa.curve.pointAdd(partial, share));
+    const u = BigInt(pShare.uu);
+    let contribChaincode = hexToBigInt(pShare.chaincode);
+    const chaincodes = [contribChaincode, ...nShares.map(({ chaincode }) => hexToBigInt(chaincode))];
+    const chaincode = chaincodes.reduce((acc, chaincode) => (acc + chaincode) % chaincodeBase);
+
+    // Verify shares.
+    for (const share of nShares) {
+      if (share.v) {
+        try {
+          Ecdsa.shamir.verify(hexToBigInt(share.u), [hexToBigInt(share.y), hexToBigInt(share.v!)], pShare.i);
+        } catch (err) {
+          throw new Error(`Could not verify share from participant ${share.j}. Verification error: ${err}`);
+        }
+      }
+    }
```

Comment:
NIT: rather than duplicating this from keycombine, I think its better to use its as a function which can be reused across both the functions